src/ripple/app/tx/TransactionEngine.cpp

```diff
@@ -241,8 +241,17 @@ TransactionEngine::applyTransaction (
                 throw std::runtime_error ("Duplicate transaction applied to closed ledger");
             }
 
-            // Charge whatever fee they specified.
-            mLedger->destroyCoins (getNValue (txn.getTransactionFee ()));
+            // Charge whatever fee they specified. We break the encapsulation of
+            // STAmount here and use "special knowledge" - namely that a native
+            // amount is stored fully in the mantissa:
+            auto const fee = txn.getTransactionFee ();
```

Comment:
What you've done is okay with me.  But you don't *have* to break encapsulation of `STAmount`.  You could add `STAmount::getNativeAmount()` that returns a `uint64_t` and throws if the amount is not Native.  You'd call that instead of `fee.mantissa()`.  Then you could remove the apology here.